core/src/HPX/Kokkos_HPX.hpp

```diff
@@ -163,9 +163,17 @@ class HPX {
   using size_type            = memory_space::size_type;
   using scratch_memory_space = ScratchMemorySpace<HPX>;
 
+// FIXME_HPX spurious warnings like
+// error: 'SR.14123' may be used uninitialized [-Werror=uninitialized]
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wuninitialized"
+
   HPX()
       : m_instance_data(Kokkos::Impl::HostSharedPtr<instance_data>(
             &m_default_instance_data, &default_instance_deleter)) {}
```

Comment:
I did a few small tests, and one of the other things that silences the warning is initializing `m_instance_data` inside the body of the constructor. I think this points to the warning indeed being bogus. I'm thinking perhaps the static analyzer gets confused by the throwing execution path of `HostSharedPtr` or something like that?

In any case, I think, if this is indeed a false positive, the `#pragma GCC diagnostic ignored` is in my opinion clearer than e.g. moving the initialization to the constructor body (or another code change that silences the warning).